data-processing/entities/terms/extractor.py

```diff
@@ -10,110 +9,169 @@
+		global matched_terms
+		global matched_definition_texts
+		global matched_definition_htmls
+		global matched_terms_start
+		global matched_terms_end
+		global matched_terms_context_str
+
+		for text_segment in plaintext_segments:
+			text_segment_string = str(text_segment.text.lower())
+
+			# Making two lists from the text_segment_string; one is to capture words without dashes
+			# The other is to capture words such as "Gradient-Activated" (but this word happens to not be
+			# in the glossary
+			text_segment_string_split_on_dashes = re.split(
+				"[-\s+]", text_segment_string)
+			text_segment_string_not_split_on_dashes = re.split(
+				"[\s+]", text_segment_string)
+
+			# Adding all terms that were both in the glossary and in the .tex file into a
+			# set so we don't get duplicates
+			matches_tuples = []
+			matches = set()
+			for word in text_segment_string_split_on_dashes:
+				for object in terms_and_def_text_html:
+
+					# Object[0] is the term
+					if object[0] == word:
+
+						# Adding term name, text_segment from which term was found, def text and def html
+						matches_tuples.append([object[0], text_segment_string, object[1], object[2]])
+
+			for word2 in text_segment_string_not_split_on_dashes:
+				for object in terms_and_def_text_html:
+					if object[0] == word2:
+						matches_tuples.append([object[0], text_segment_string, object[1], object[2]])
+			for object in terms_and_def_text_html:
+				term_list = re.split("\s+", object[0])
+				if len(term_list) > 1:
+					if object[0] in text_segment_string:
+						matches_tuples.append([object[0], text_segment_string, object[1], object[2]])
+
+			for i in matches_tuples:
+				list_to_add = i[:-1]
+				if isinstance(i[-1], list):
+					last_elem = tuple(i[-1])
+				else:
+					last_elem = i[-1]
+				list_to_add.append(last_elem)
+				tuple_to_add = tuple(list_to_add)
+				matches.add(tuple_to_add)
+
+			###Getting the start and end indices of the matched terms
+			for match in matches:
+				start_index_of_string = text_segment_string.find(match[0])
+				end_index_of_string = start_index_of_string + len(match[0]) - 1
+				matched_terms.append(match[0])
+				matched_terms_context_str.append(match[1])
+				matched_definition_texts.append(match[2])
+				matched_definition_htmls.append(match[3])
+				matched_terms_start.append(text_segment.tex_start + start_index_of_string)
+				matched_terms_end.append(text_segment.tex_start + end_index_of_string)
+
+		def duplicates(lst, item):
+			return [i for i, x in enumerate(lst) if x == item]
+
+		# To Do:
+		# 1) Determine the objects that share the same start index.
+		# 2) Keep the object with the largest end index.
+		# 3) Remove all other objects from all lists.
+
+		def delete_duplicate_start(term, start, end, context_str):
+			for start_index in start:
+
+				# 1)
+				duplicate_indices = duplicates(start, start_index)
+
+				# 2) Keep the object with the largest end index.
+				if len(duplicate_indices) > 1:
+					print("OH NO")
```

Comment:
If you want to keep this message, can this use `logging.warning` instead (see these other examples in the code https://github.com/allenai/scholar-reader/search?q=logging.warning&unscoped_q=logging.warning), and change the message to clearly indicate what's going wrong?

Reply:
Removed this debugging print statement